pkg/network/server.go

```diff
@@ -715,10 +715,29 @@ func (s *Server) handleGetHeadersCmd(p Peer, gh *payload.GetBlockByIndex) error
 	return p.EnqueueP2PMessage(msg)
 }
 
-// handleConsensusCmd processes received consensus payload.
-// It never returns an error.
-func (s *Server) handleConsensusCmd(cp *consensus.Payload) error {
-	s.consensus.OnPayload(cp)
+const extensibleVerifyMaxGAS = 2000000
+
+// handleExtensibleCmd processes received extensible payload.
+func (s *Server) handleExtensibleCmd(e *payload.Extensible) error {
+	if err := s.chain.VerifyWitness(e.Sender, e, &e.Witness, extensibleVerifyMaxGAS); err != nil {
```

Comment:
Isn't it duplicating signature check in `validatePayload`?

Reply:
Yes, fixed `validatePayload`.